### PR TITLE
Feat/153: resource `apstra_datacenter_generic_system`

### PR DESCRIPTION
**Pull request by chrismarget-j**
Introduce `apstra_datacenter_generic_system` resource:

```hcl
resource "apstra_datacenter_generic_system" "example" {
  blueprint_id      = "blueprint-id-goes-here"
  name              = "my generic system"
  hostname          = "mygenericsystem.company.com"
  tags              = ["terraform", "test"]
  links = [
    {
      tags                          = ["10G", "terraform", "bond0"]
      lag_mode                      = "lacp_active"
      target_switch_id              = "switch-id-goes-here"
      target_switch_if_name         = "xe-0/0/6"
      target_switch_if_transform_id = 1
      group_label                   = "bond0"
    },
    {
      tags                          = ["10G", "terraform", "bond0"]
      lag_mode                      = "lacp_active"
      target_switch_id              = "switch-id-goes-here"
      target_switch_if_name         = "xe-0/0/6"
      target_switch_if_transform_id = 1
      group_label                   = "bond0"
    },
  ]
}
```

### CODE REVIEW

File: apstra/blueprint/datacenter_generic_system.go [hunk 1]
```diff
@@ -0,0 +1,569 @@
...
+	}
+
+	o.Tags = utils.SetValueOrNull(ctx, types.StringType, tags, diags)
+}
+
+func (o *DatacenterGenericSystem) ReadLinks(ctx context.Context, bp *apstra.TwoStageL3ClosClient, diags *diag.Diagnostics) {
+	// Extract the prior state into a map of links keyed by endpoint digest.
+	// We need quick access to the prior data so that we know whether to
+	// populate `group_label` into our result. The `group_label` field is an
+	// "Optional" but NOT "Computed" attribute because of the complexity of
```
Comment (rajagopalans) [inline apstra/blueprint/datacenter_generic_system.go:147]: need clarification here.

Comment (chrismarget-j) [reply apstra/blueprint/datacenter_generic_system.go]: Does 120d119 help?

File: apstra/resource_datacenter_generic_system.go [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+		resp.Diagnostics.AddError("failed to determine new generic system ID from links", err.Error())
+	}
+	plan.Id = types.StringValue(genericSystemId.String())
+
+	// pull Apstra-generated strings if not specified by the user
+	if plan.Name.IsUnknown() || plan.Hostname.IsUnknown() {
+		err = plan.GetLabelAndHostname(ctx, bp)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				fmt.Sprintf("error parsing labels from new generic server %s", plan.Id), err.Error())
```
[the remark this answers is not in the document]
Comment (rajagopalans) [reply apstra/resource_datacenter_generic_system.go:97]: Also starting with error.

[the remark this answers is not in the document]
Comment (chrismarget-j) [reply apstra/resource_datacenter_generic_system.go]: several improved errors in c5e9efe

**Review by rajagopalans — APPROVED** [review]

**Review by rajagopalans — APPROVED** [review]

**Pull request merged**